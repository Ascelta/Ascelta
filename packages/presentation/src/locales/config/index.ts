import { initReactI18next } from 'react-i18next';
import { getLocales } from 'react-native-localize';
import i18n from 'i18next';
import en from './../en.json';
import ja from './../ja.json';

export const supportedLngs = {
  en: 'English',
  ja: '日本語',
};

i18n.use(initReactI18next).init({
  supportedLngs: Object.keys(supportedLngs),
  resources: {
    en: { translation: en },
    ja: { translation: ja },
  },
  lng: getLocales()[0].languageCode,
  fallbackLng: ['en', 'ja'],
  interpolation: {
    escapeValue: false,
  },
});

export default i18n;
