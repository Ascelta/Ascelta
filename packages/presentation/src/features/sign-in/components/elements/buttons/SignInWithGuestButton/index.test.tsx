import { defaultConfig } from '@tamagui/config/v4';
import '@testing-library/jest-dom';
import { act, fireEvent, render, screen } from '@testing-library/react';
import { TamaguiProvider, createTamagui } from 'tamagui';
import { SignInWithGuestButton } from './index.tsx';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

jest.mock('../../../../../../contexts', () => {
  // モックファクトリ内で直接作る
  const mockSignIn = jest.fn();
  return {
    __esModule: true,
    useAuth: jest.fn().mockImplementation(() => ({
      signIn: mockSignIn,
    })),
    mockSignIn: mockSignIn,
  };
});

const mockSignIn = require('../../../../../../contexts').mockSignIn;

describe('<SignInWithGuestButton />', () => {
  const config = createTamagui(defaultConfig);
  const setup = () => {
    return render(
      <TamaguiProvider config={config}>
        <SignInWithGuestButton />
      </TamaguiProvider>,
    );
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('初期表示時のスナップショットが一致すること', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  it.skip(`ボタンをタップした場合は、確認ダイアログが表示されて、「キャンセル」ボタンをタップすると signIn が呼ばれないこと`, async () => {
    setup();
    const continueWithGuestButtonText = screen.getByText('CONTINUE_WITH_GUEST');
    const guestSignInAlertTitle = screen.queryByText('GUEST_SIGN_IN_ALERT_TITLE');
    const guestSignInAlertMessage = screen.queryByText('GUEST_SIGN_IN_ALERT_MESSAGE');
    const cancelButtonText = screen.queryByText('CANCEL');
    const continueButtonText = screen.queryByText('CONTINUE');
    expect(continueWithGuestButtonText).toBeInTheDocument();
    expect(guestSignInAlertTitle).not.toBeInTheDocument();
    expect(guestSignInAlertMessage).not.toBeInTheDocument();
    expect(cancelButtonText).not.toBeInTheDocument();
    expect(continueButtonText).not.toBeInTheDocument();

    await act(async () => {
      fireEvent.click(continueWithGuestButtonText);
    });
    expect(continueWithGuestButtonText).toBeInTheDocument();
    expect(guestSignInAlertTitle).toBeInTheDocument();
    expect(guestSignInAlertMessage).toBeInTheDocument();
    expect(cancelButtonText).toBeInTheDocument();
    expect(continueButtonText).toBeInTheDocument();

    await act(async () => {
      fireEvent.click(cancelButtonText);
    });
    expect(continueWithGuestButtonText).toBeInTheDocument();
    expect(guestSignInAlertTitle).not.toBeInTheDocument();
    expect(guestSignInAlertMessage).not.toBeInTheDocument();
    expect(cancelButtonText).not.toBeInTheDocument();
    expect(continueButtonText).not.toBeInTheDocument();
    expect(mockSignIn).not.toHaveBeenCalled();
  });
});
