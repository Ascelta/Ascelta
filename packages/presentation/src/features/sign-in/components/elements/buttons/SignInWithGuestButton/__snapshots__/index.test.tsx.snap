// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInWithGuestButton /> 初期表示時のスナップショットが一致すること 1`] = `
<DocumentFragment>
  <span
    class=""
    style="display: contents;"
  >
    <span
      class="t_sub_theme t_Button _dsp_contents is_Theme"
    >
      <span
        aria-controls="content-:r2:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="is_Button _bg-0hover-backgroundH3423444 _btc-0hover-borderColor69916956 _brc-0hover-borderColor69916956 _borderBottomColor-0hover-borderColor69916956 _borderLeftColor-0hover-borderColor69916956 _bg-0active-backgroundP3496915 _btc-0active-borderColor77378595 _brc-0active-borderColor77378595 _borderBottomColor-0active-borderColor77378595 _borderLeftColor-0active-borderColor77378595 _outlineColor-0focus-visible-outlineColo114 _outlineStyle-0focus-visible-solid _outlineWidth-0focus-visible-2px _dsp-flex _items-center _flexBasis-auto _bxs-border-box _pos-relative _minH-0px _minW-0px _shrink-0 _flexDirection-row _pr-c-space-tru101 _pl-c-space-tru101 _height-c-size-true _btlr-c-radius-6 _btrr-c-radius-6 _borderBottomRightRadius-c-radius-6 _borderBottomLeftRadius-c-radius-6 _justify-center _fwr-nowrap _cur-pointer _bg-background _btw-1px _brw-1px _borderBottomWidth-1px _borderLeftWidth-1px _btc-transparent _brc-transparent _borderBottomColor-transparent _borderLeftColor-transparent _borderBottomStyle-solid _borderTopStyle-solid _borderLeftStyle-solid _borderRightStyle-solid _width-10037"
        data-disable-theme="true"
        data-state="closed"
        role="button"
        tabindex="0"
      >
        <div
          class="_dsp-flex _flexBasis-auto _bxs-border-box _pos-relative _minH-0px _minW-0px _shrink-0 _flexDirection-row _width-10037 _items-center _justify-center"
        >
          <p
            class="is_Paragraph font_body _dsp-inline _bxs-border-box _ww-break-word _mt-0px _mr-0px _mb-0px _ml-0px _ff-f-family _fw-bold _ls-f-letterSpa1360334201 _fs-f-size-5 _lh-f-lineHeigh112924 _col-color _select-auto _ws-normal"
          >
            CONTINUE_WITH_GUEST
          </p>
        </div>
      </span>
    </span>
    <div
      style="display: contents;"
    />
  </span>
</DocumentFragment>
`;
